src/Maestro/Maestro.MergePolicies/VersionDetailsPropsMergePolicy.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using Maestro.MergePolicyEvaluation;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.Helpers;
+
+#nullable enable
+namespace Maestro.MergePolicies;
+public class VersionDetailsPropsMergePolicy : MergePolicy
+{
+    public override string Name => "VersionDetailsProps";
+
+    public override string DisplayName => "Version Details Properties Merge Policy";
+
+    public override async Task<MergePolicyEvaluationResult> EvaluateAsync(PullRequestUpdateSummary pr, IRemote remote)
+    {
+        if (pr.CodeFlowDirection == CodeFlowDirection.ForwardFlow)
+        {
+            // TODO: https://github.com/dotnet/arcade-services/issues/4998 Make the check work for forward flow PRs once we implement the issue
+            return SucceedDecisively("Version Details Properties Merge Policy: Not a backflow PR");
+        }
+
+        XmlDocument versionDetailsProps;
+        try
+        {
+            versionDetailsProps = DependencyFileManager.GetXmlDocument(await remote.GetFileContentsAsync(
+                VersionFiles.VersionDetailsProps,
+                pr.TargetRepoUrl,
+                pr.HeadBranch));
+        }
+        catch (DependencyFileNotFoundException)
+        {
+            // TODO: this should only be in for the transition period until we add VersionDetailsProps to all codeflow repos
+            return SucceedDecisively("Version Details Properties Merge Policy: VersionDetailsProps file not found, skipping validation.");
+        }
+        catch
```

Comment:
I wonder if we should catch XML exceptions when the file can't be parsed to let people know that it's corrupted?

Reply:
fair point, adding it